[Shipping labels] Update shipments tab

irfano opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->
Part of WOOMOB-580

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR adds a fading effect at the end of the shipments tabs and a divider at the bottom of the tabs, based on the design.

### Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->
1. Go to Orders.
2. Select an order that has multiple shipments.

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->
Screens below.

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
|Before|After|
|-|-|
|<img width="1344" height="2992" alt="before" src="https://github.com/user-attachments/assets/a42903bb-bb45-4846-b270-950dd5c62f0e" />|<img width="1344" height="2992" alt="after" src="https://github.com/user-attachments/assets/ddc94d63-ad09-4410-9bd9-874d4152139f" />|
|<img width="1344" height="2992" alt="before-dark" src="https://github.com/user-attachments/assets/3a1cac47-d73d-4126-adfc-c57000176e52" />|<img width="1344" height="2992" alt="after-dark" src="https://github.com/user-attachments/assets/43526bda-b7ad-4293-af8a-042c51764586" />|

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

irfano committed:
Add horizontal fade to shipments tabs

irfano committed:
Add a divider at the bottom of the Shipments tab

irfano committed:
Migrate ShipmentsTabRow to Material 3

irfano committed:
Update titleSmall lineHeight in Compose theme

The lineHeight for the titleSmall text style in the Compose theme typography has been changed from 24.sp to 20.sp.

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

irfano committed:
Update release notes for shipments tab layout

The release notes are updated to reflect the layout changes in the shipments tab.

irfano commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/theme/Typography.kt:
```diff
@@ -148,7 +148,7 @@ val Material3Typography = Typography(
         fontWeight = FontWeight.Medium,
         fontSize = 14.sp,
-        lineHeight = 24.sp
+        lineHeight = 20.sp
```
`titleSmall` was not being used in the app. I updated its line height to match the default lineHeight of the `titleSmall` style from [Material 3 theme](https://m3.material.io/styles/typography/type-scale-tokens#a734c6ed-634c-4abb-adb2-35daf0aed06a).

irfano committed:
Merge branch 'trunk' into issue/WOOMOB-580-add-fadeing-edge-to-shipments-tab

JorgeMucientes reviewed (approved):
Thanks for addressing this @irfano

Given that you've addressed this UI improvements why not apply the same changes to the tabs in the split shipment screen?

<img width="281" height="249" alt="Screenshot 2025-08-11 at 18 33 01" src="https://github.com/user-attachments/assets/d5a1ba36-62d4-40b3-b1c4-e6ade208ddd7" />

irfano committed:
Merge branch 'trunk' into issue/WOOMOB-580-add-fadeing-edge-to-shipments-tab

irfano committed:
Move fading edge modifier to ShipmentsTabRow

irfano committed:
Add divider to split shipment screen

A `HorizontalDivider` is added between the tab row and the horizontal pager in the split shipment screen.

irfano commented:
Good catch, @JorgeMucientes! I overlooked that. I’ve also applied the changes to the split shipment screen.
